kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3WindowTable.java

```diff
@@ -0,0 +1,226 @@
+          name,
+          kafkaPartition,
+          lssId.id()
+      ));
+    }
+  }
+
+  @Override
+  public byte[] fetch(
+      final int kafkaPartition,
+      final Bytes key,
+      final long windowStart
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    final int pssId = pssPartitioner.pss(key.get(), this.lssId);
+    final var windowKey = new WindowedKey(key, windowStart);
+    final var windowKeyBytes = keySerde.serialize(windowKey);
+
+    return rs3Client.get(
+        storeId,
+        lssId,
+        pssId,
+        flushManager.writtenOffset(pssId),
+        windowKeyBytes
+    ).orElse(null);
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> fetch(
+      final int kafkaPartition,
+      final Bytes key,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> backFetch(
+      final int kafkaPartition,
+      final Bytes key,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> fetchRange(
+      final int kafkaPartition,
+      final Bytes fromKey,
+      final Bytes toKey,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> backFetchRange(
+      final int kafkaPartition,
+      final Bytes fromKey,
+      final Bytes toKey,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> fetchAll(
+      final int kafkaPartition,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public KeyValueIterator<WindowedKey, byte[]> backFetchAll(
+      final int kafkaPartition,
+      final long timeFrom,
+      final long timeTo
+  ) {
+    throwIfPartitionNotInitialized(kafkaPartition);
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  public WalEntry insert(
+      final int kafkaPartition,
+      final WindowedKey key,
+      final byte[] value,
+      final long timestampMs
+  ) {
+    return new Put(keySerde.serialize(key), value);
+  }
+
+  @Override
+  public WalEntry delete(final int kafkaPartition, final WindowedKey key) {
+    return new Put(keySerde.serialize(key), null);
+  }
+
+  @Override
+  public long fetchOffset(final int kafkaPartition) {
```

Comment:
not really related to this PR but `fetchOffset` is kind of a vague/misleading name for this API. Maybe we should rename to something like `lastWrittenOffset` or `restorationStartOffset`, etc?

Reply:
Agreed. I opened a spearate PR https://github.com/responsivedev/responsive-pub/pull/444.